src/test/java/com/deque/html/axecore/selenium/AxeIntegrationUnitTest.java

```diff
@@ -119,6 +125,93 @@ public void runScanOnGivenElementChrome() throws IOException, OperationNotSuppor
     Assert.assertEquals(3, results.getViolations().size());
   }
 
+  @Test()
+  public void htmlReportFullPage() throws IOException, ParseException {
+    String path = createReportPath();
+    HtmlReporter.createAxeHtmlReport(this.webDriver, path);
+    validateReport(path, 5, 46, 0, 57);
+
+    File file = new File(path);
+
+    if (file.exists()) {
+      Assert.assertTrue("File was not deleted", file.delete());
+    }
+  }
+
+  @Test()
+  public void htmlViolationsOnlyReportFullPage() throws IOException, ParseException {
+    String path = createReportPath();
+    HtmlReporter.createAxeHtmlViolationsReport(this.webDriver, path);
+
+    String text = Files.lines(Paths.get(path), StandardCharsets.UTF_8)
+        .collect(Collectors.joining(System.lineSeparator()));
+    Document doc = Jsoup.parse(text);
+
+    // Check violations
+    String xpath = "#ViolationsSection > div > div.htmlTable";
+    Elements liNodes = doc.select(xpath) != null ? doc.select(xpath) : new Elements();
+    Assert.assertEquals("Expected " + 5 + " violations", 5, liNodes.size());
+
+    File file = new File(path);
+
+    if (file.exists()) {
+      Assert.assertTrue("File was not deleted", file.delete());
+    }
+  }
+
+  @Test()
+  public void htmlReportOnElement() throws IOException, ParseException {
+    String path = createReportPath();
+    HtmlReporter.createAxeHtmlReport(this.webDriver, this.webDriver.findElement(By.cssSelector("main")), path);
+    validateReport(path, 3, 16, 0, 69);
+
+    File file = new File(path);
+
+    if (file.exists()) {
+      Assert.assertTrue("File was not deleted", file.delete());
+    }
+  }
+
+  @Test
+  public void reportRespectRules() throws IOException, ParseException {
+    String path = createReportPath();
+    AxeBuilder builder = new AxeBuilder().disableRules(Collections.singletonList("color-contrast"));
+    HtmlReporter.createAxeHtmlReport(webDriver, builder.analyze(webDriver), path);
+    validateReport(path, 4, 39, 0, 57);
+
+    File file = new File(path);
+
+    if (file.exists()) {
+      Assert.assertTrue("File was not deleted", file.delete());
+    }
+  }
+
+  @Test
+  public void ReportSampleResults() throws IOException, ParseException {
+    String path = createReportPath();
+    HtmlReporter.createAxeHtmlReport(webDriver, path);
+    validateReport(path, 5, 46, 0, 57);
+
+    String text = new String(Files.readAllBytes(Paths.get(path)));
+    Document doc = Jsoup.parse(text);
+
+    String reportContext = doc.selectFirst("#reportContext").text();
+    Assert.assertTrue(reportContext.contains("Url: ")
+        && reportContext.contains(integrationTestTargetFile.toURI().toURL().getPath()));
+    Assert.assertTrue(reportContext.contains("Orientation: landscape-primary"));
+    Assert.assertTrue(reportContext.contains("Size: 1920 x 1200"));
+    // TODO: assert time contains the time scan is taken

```

Comment:
Unless we have a constant for the time, it seems that this assertion will have to be deleted from the unit test. 
Let me know your thoughts.